core/domain/base_model_validators.py

```diff
@@ -84,10 +86,23 @@ def __init__(
             model_class: ClassObject. The external model class.
             model_ids: list(str). The list of external model ids to fetch the
                 external models.
-        """
+            remove_system_user_ids: bool. Whether to remove the system user IDs
+                from the model_ids.
+            remove_pseudonymous_ids: bool. Whether to remove pseudonymous user
+                IDs from the model_ids.
+        """
+        filtered_model_ids = model_ids
+        if remove_system_user_ids:
+            filtered_model_ids = list(
+                set(filtered_model_ids) - set(feconf.SYSTEM_USERS.values()))
```

Comment:
This is a general class, right? Why are we assuming this to be a user_id?

I would suggest:
1. Either changes the validation method to highlight these stuff i.e, it should not validate user_models if the user_id is system_id or pseudonymous. OR have this method check for the class name first* I would prefer the first one as that's much cleaner and better. (I can help with a better suggestion if I'll have a context on why this removal is needed.)

Reply:
This validation is needed for several fields which consists of user_ids. These user_ids sometimes have oppia bot or pseudo user ids which do not have a user setting model. Thus for these ids user setting model can not be fetched thus causing a validation error. Since these cannot be fetched, these should be removed in the fetcher model. The remove_system_user_ids and the remove_pseudonymous_ids are set to true only if the attributes contain the user ids and need to be avoided. Since this is a common fetcher, removing it here would decrease a lot of code also.

Reply:
I guess we can have user specific fetcher that will have these attributes and will be used only for retrieving `UserSettingsModel`.

Reply:
@rohitkatlaa, is this resolved?

Reply:
I am sorry for the delay, I have now added a new class specifically for userSettingsModel.